### ISSUE TITLE
xunit.runner.visualstudio - GetAvailableRunnerReporters scans too many dlls

### ISSUE THREAD

**Issue by tapika** [xunit/xunit#2200]
This concerns to xunit v2 branch unit test.

`src\xunit.runner.visualstudio\VsTestRunner.cs` gets dll's using `*.dll`.
```
            foreach (var dllFile in Directory.GetFiles(runnerPath, "*.dll").Select(f => Path.Combine(runnerPath, f)))
```

and for example with one application, which I'm testing this results in 230 dlls, and loading all of them is not possible - I get approximately 35 `System.BadImageFormatException` exceptions, because some of .dll's are native dll, cannot be loaded as assembly, some apparently are 32-bit, loaded from 64-bit application - also not possible.

By quick check what and where resides - I've changed filter to `"*runner*.dll"` - after that debugging start works faster, no exceptions, and find finds only xunit dll's, not everything.

I'm not sure about what is currently happening for `v2` versus `master` on xunit, but can provide pull request for `v2` xunit.

**Comment by tapika** [comment]
After my change I still see couple of c++ exceptions
`Microsoft C++ exception: EEFileLoadException at memory location `
when loading for example `xunit.runner.reporters.net452.dll`.

Maybe missing some of compiled dll's, need to try out whole nuget package build.

**Comment by Enigmatrix** [comment]
Is there any fix for this currently? I was debugging a unit test and one of the project deps pulled in a native .dll. The debugger broke at VsTestTunner, probably due to [the inclusion of all *.dll](https://github.com/xunit/visualstudio.xunit/blob/main/src/xunit.runner.visualstudio/VsTestRunner.cs#L706) to be loaded by `Assembly.Load`. 

![image](https://user-images.githubusercontent.com/12296254/132087902-f6e58129-febe-4d15-b678-517fd2b29669.png)

`exports.dll` is the name of my native library

**Comment by bradwilson** [comment]
Sorry for getting back to this late.

This was fixed in [v2 Core Framework `2.5.0`](https://xunit.net/releases/v2/2.5.0) and [Visual Studio Adapter `2.5.0`](https://xunit.net/releases/visualstudio/2.5.0). The fix is part of `xunit.runner.utility` so any third party runners must be rebuilt against the 2.5.0 release of that library (or later) to pick up the change, as well as relying on the new utility function which does the work:

https://github.com/xunit/xunit/blob/280a0cbf6a4e1d11f4108fbe2b811abd4c9eab76/src/xunit.runner.utility/Reporters/RunnerReporterUtility.cs#L17-L24
